eng/slngen.targets

```diff
@@ -5,10 +5,7 @@
     <SlnGenSolutionFolder Condition="'$(SlnGenSolutionFolder)' == ''">src</SlnGenSolutionFolder>
 
     <!-- Don't include shared framework projects referenced via a named reference except the leaf's project. -->
-    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' != '$(SlnGenMainProject)'">false</IncludeInSolutionFile>
-
-    <!-- Include CompilerServices.Unsafe as it's a transitive test dependency. -->
-    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' == 'System.Runtime.CompilerServices.Unsafe'">true</IncludeInSolutionFile>
+    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' != '$(SlnGenMainProject)' and '$(TargetFrameworks)' == '$(NetCoreAppCurrent)'">false</IncludeInSolutionFile>
```

Comment:
can you elaborate on how we intend to use ```'$(TargetFrameworks)' == '$(NetCoreAppCurrent)'```

Reply:
I'm not sure what you mean by "how we intent do use ..."? We don't want to include reference projects in the solution file which only target NetCoreAppCurrent as those are currently referenced via named references by the source project and hence don't need to be part of the solution file.

Reply:
Added a code comment which should clarify.